### PR TITLE
Update 'github-pages.yaml' Workflow - Fix several warnings & enable Caching pip packages

### LINKED ISSUE #2417: GitHub Pages Workflow Shows a Warning on the Caching Step
The `actions/cache` Step in `github-pages.yml` Workflow shows a warning message with the follow:
```
Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.
```

### PR DESCRIPTION
**Pull request by og-mrk**
## Type of Change
- [x] Refactoring
- [x] CI/CD improvement

## Description
This PR Makes `github-pages.yaml` Workflow more maintainable/readable, also makes `git-revision-date-localized-plugin` plugin work by pull all commit history (see references for more info), and make `github-pages.yaml` generate no warnings*.. ~_Except for `nodejs` warning_, which can be resolved by updating `actions/setup-python` to `v5`.. but it can't be done at the moment, as it's pinned to use `v4` or less. The reason behind this version pinning is unclear...~ Fixed in [this commit](https://github.com/ChrisTitusTech/winutil/commit/8ae0bc6a6c7cef95ee59db55ef5f9c03895ca2e4), see my review comments on why I've updated this action.

Nonetheless, these changes should make `github-pages.yaml` workflow cache the results (downloads) of pip packages, and these packages are specified in the new `requirements.txt` file, which's a common file used in Python Environments, which specifies every package to install, and what version to use, using `PackageName==SpecificVersion` Syntax.

<details><summary>How to make <code>requirements.txt</code> File (Step By Step)</summary>
<p>

</br>
</br>

</br>
</br>

</p>
</details> 

## Testing
Ran this GitHub Action manually on this PR Branch, and it works as expected.

> [!NOTE]
> The first run would take much longer then next runs.. as it's compressing & using GitHub's Built-in functionality of Caching between Workflow Runs, _more info can be found in references below_.

## Impact
Won't have a much impact on building/deploying the Docs website.. it'll shave a few seconds here & there, but IMHO it's better than nothing, and lowering Runtime Usage is always a good thing.

## Issue related to PR
- Resolves #2417

## Additional Information (References)
- [Fetch all history for all tags and branches - actions/checkout](https://github.com/actions/checkout/blob/main/README.md#Fetch-all-history-for-all-tags-and-branches)
- [About caching workflow dependencies - GitHub Docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#about-caching-workflow-dependencies)
- [Caching packages dependencies - actions/python-setup](https://github.com/actions/setup-python/blob/main/README.md#caching-packages-dependencies)
- [Python Requirements.txt – How to Create and Pip Install Requirements.txt in Python - freecodecamp.org](https://www.freecodecamp.org/news/python-requirementstxt-explained/)
- [How do I remove all packages installed by pip? - StackOverFlow](https://stackoverflow.com/questions/11248073/how-do-i-remove-all-packages-installed-by-pip)
- [How to properly disable warnings and set logging to error for every library mkdocs uses? - mkdocs/mkdocs GitHub Discussion](https://github.com/mkdocs/mkdocs/discussions/3581)

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by og-mrk** [comment]
@ChrisTitusTech 
image

File: .github/workflows/github-pages.yaml [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
-      - uses: actions/setup-python@v4
+      - name: Checkout Repository
+        uses: actions/checkout@v4
         with:
-          python-version: 3.x
-      - uses: actions/cache@v4
+          fetch-depth: '0'  # Fetch all commit history for all branches as well as tags.
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
```
Comment (og-mrk) [inline .github/workflows/github-pages.yaml:22]: Reason behind upgrading `actions/setup-python` to `v5` is pretty simple.. it just works after upgrading this action, and I can confirm this after testing it in a new branch.

I've branched-out & committed/pushed a [very simple commit](https://github.com/og-mrk/winutil/commit/c409409a0dc1a0a2d662710170cb5fcb7fc2d036), which even after deleting the branch, still exists thanks to GitHub having everything stay even after "Deleting" it, which's an awesome feature for some people.. _but somewhat terrifying to others_, but nonetheless.. it does work & produces no errors/fails to build & deploy the docs website, [link to the run](https://github.com/og-mrk/winutil/actions/runs/10302979965).

File: .github/dependabot.yml [hunk 2]
```diff
@@ -7,5 +7,3 @@ updates:
     ignore:
       - dependency-name: "actions/stale"
         versions: '>= 9'
-      - dependency-name: "actions/setup-python"
-        versions: '> 4'
```
Comment (og-mrk) [inline .github/dependabot.yml:11]: Please read [this review comment](https://github.com/ChrisTitusTech/winutil/pull/2539/files#r1709526756) to understand why I've removed this Dependabot Rule.

File: requirements.txt
Comment (og-mrk) [inline requirements.txt:1]: Specifying each and every dependency & sub-dependency (and even sub-sub-dependency) is a great thing IMO, as not only it shows what we install.. but also what VERSION we should be using, of course this's mainly used as setup file, you run `pip install -r requirements.txt` and you're off to the races!

If you want to update or add a new package (which may or may require dependencies).. you can search online on how to properly update this file, I've done it with the most simple & straight forward way.. which's uninstalling every package, clearing the cache, and then install the packages required to build & deploy the docs website, see PR description for more details on the steps, and which "main" packages I've installed, which'll generate a similar list like the one below.

**Review by ChrisTitusTech — APPROVED** [review]
A great idea so dependabot doesn't screw us on a bad update. Thanks @og-mrk

**Pull request merged**